### PR TITLE
TS migrate: add events

### PR DESCRIPTION
**Pull request by scottrippey**
# What
- Migrates the final part of `victory-utils`: `add-events`!
- With this conversion, we can now migrate other Victory components, like `victory-line`.

### CODE REVIEW

File: packages/victory-core/src/victory-util/add-events.tsx [hunk 1]
```diff
@@ -288,24 +361,28 @@ export default (WrappedComponent, options) => {
...
 
-    animateComponent(props, defaultAnimationWhitelist) {
+    animateComponent(props: TProps, defaultAnimationWhitelist: string[]) {
       const animationWhitelist =
-        props.animate && props.animate.animationWhitelist
+        typeof props.animate === "object" &&
+        props.animate &&
+        props.animate.animationWhitelist
           ? props.animate.animationWhitelist
           : defaultAnimationWhitelist;
```
Comment (gksander) [inline packages/victory-core/src/victory-util/add-events.tsx:370]: Might consider `props.animate?.animationWhitelist || defaultAnimationWhitelist`

File: packages/victory-core/src/victory-util/add-events.tsx [hunk 2]
```diff
@@ -25,34 +28,104 @@ const datumHasXandY = (datum) => {
...
-    }
 
-    shouldComponentUpdate(nextProps) {
+      this.cacheValues(this.getCalculatedValues(props));
+    }
+    state = {};
+    getEventState = Events.getEventState.bind(this);
+    getScopedEvents = Events.getScopedEvents.bind(this);
+    getEvents = (p, target, eventKey) => {
+      return Events.getEvents.call(
```
Comment (gksander) [inline packages/victory-core/src/victory-util/add-events.tsx:112]: `.call`, now that's a method I haven't seen in awhile 🤩

**Review by gksander — APPROVED** [review]
Nice!

File: packages/victory-core/src/victory-util/add-events.tsx (hunk 2 above)
Comment (scottrippey) [reply packages/victory-core/src/victory-util/add-events.tsx:112]: I know ... as I was migrating these `Events.*` methods, I was surprised to use the `this` argument typing again.  Not very common nowadays!

File: packages/victory-core/src/victory-util/add-events.tsx (hunk 1 above)
Comment (scottrippey) [reply packages/victory-core/src/victory-util/add-events.tsx]: Done

**Pull request merged**